src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/evolutionaryAlgorithm/EvolutionaryAlgorithmQueryOptimizer.java

```diff
@@ -46,9 +47,10 @@ public PhysicalPlan optimize( final PhysicalPlan plan ) throws QueryOptimization
         // initialize the first generation
         List<PhysicalPlanWithCost> currentGen = generateFirstGen( plan, ruleApplicationCache );
 
+        List<List<PhysicalPlanWithCost>> previousGenerations = new ArrayList<>();
         // TODO: extend this implementation with different sorts of termination criteria
-        // termination criteria: number of generations
-        for ( int gen = 1; gen < nmSteps; gen++ ){
+        while( !(terminateCriterion.readyToTerminate( nmSteps, currentGen, previousGenerations)) ){
```

Comment:
```suggestion
        while( ! terminateCriterion.readyToTerminate(nmSteps, currentGen, previousGenerations) ) {
```

Reply:
no need for the extra pair of parentheses

Reply:
Something else that's wrong here is that you should not give `nmSteps` as the first argument for `readyToTerminate`but, instead, you should give the number of the current generation.